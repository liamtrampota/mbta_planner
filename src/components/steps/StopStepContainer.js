import React from 'react';
import '../../styles/StepContainer.css';

 const StopStepContainer = ({ selectedStop, selectStop, stops }) => (
  <div>
    <h2 style={{ textAlign: 'center' }}>
      Select your stop
    </h2>
    { stops ?
      <StopStep stops={stops} selectedStop={selectedStop} selectStop={selectStop} />
      :
      <h4>
        Loading
      </h4>
    }
  </div>
)

const StopStep = ({ stops, selectedStop, selectStop }) => (
  <div className="stepContainer">
    {stops.map(stop => <Stop key={stop.id} stop={stop} selectedStop={selectedStop} selectStop={selectStop} />)}
  </div>
)

const Stop = ({ stop, selectedStop, selectStop }) => (
  <button style={{ display: 'flex', flexDirection:'column', alignItems: 'center', margin: '5px', width: '400px', backgroundColor: selectedStop && selectedStop.id===stop.id ? 'lightgreen' : '' }} onClick={()=>selectStop(stop)}>
    <h4 style={{marginBottom: '5px'}}>
      {stop.attributes.name}
    </h4>
    <p style={{marginTop: '5px'}}>
      {stop.attributes.address}
    </p>
  </button>
)

export default StopStepContainer;
