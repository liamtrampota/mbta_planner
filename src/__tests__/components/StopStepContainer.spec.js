import React from 'react';
import { shallow } from 'enzyme';
import StopStepContainer from '../../components/steps/StopStepContainer';

const props = {
  selectStop: jest.fn(),
  stops: [{ id: '1', attributes: { name: 'Ashmont', address: '21 Just Road'} }]
}
describe('StopStepContainer', () => {
  test('renders StopStepContainer', () => {
    const wrapper = shallow(<StopStepContainer {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
