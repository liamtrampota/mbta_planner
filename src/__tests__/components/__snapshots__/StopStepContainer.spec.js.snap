// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StopStepContainer renders StopStepContainer 1`] = `ShallowWrapper {}`;
